R/tidiers_ml_unsupervised_models.R

```diff
@@ -0,0 +1,62 @@
+#' Tidying methods for Spark ML unsupervised models
+#'
+#' These methods summarize the results of Spark ML models into tidy forms.
+#'
+#' @param x a Spark ML model.
+#' @param ... extra arguments (not used.)
+#' @name ml_unsupervised_tidiers
+NULL
+
+#' @rdname ml_unsupervised_tidiers
+#'
+#' @export
+tidy.ml_model_kmeans <- function(x,
+                                 ...){
+  model <- x$model
+  center <- x$centers
+
+  stats <- c("cluster_sizes", "k")
+  statistics <- stats %>%
+    lapply(function(x) ml_summary(model, x, allow_null = TRUE))
+
+  cbind(center,
+        sizes = statistics[[1]],
+        cluster = 1:statistics[[2]]) %>%
+    dplyr::as_tibble()
+}
+
+#' @rdname ml_unsupervised_tidiers
+#' @param newdata a tbl_spark of new data to use for prediction.
+#'
+#' @importFrom rlang sym
+#'
+#' @export
+augment.ml_model_kmeans <- function(x, newdata = NULL,
+                                    ...){
+
+  # if the user doesn't provide a new data, this funcion will
+  # use the training set
+  if (is.null(newdata)){
+    newdata <- x$dataset
+  }
+
+  ml_predict(x, newdata) %>%
+    dplyr::select(-!!sym("label"), -!!sym("features")) %>%
+    dplyr::mutate(prediction = !!sym("prediction") + 1) %>%
+    dplyr::rename(.cluster = !!"prediction")
+
+}
+
+#' @rdname ml_unsupervised_tidiers
+#' @export
+glance.ml_model_kmeans <- function(x,
+                                   ...) {
+
+  max.iter <- x$pipeline_model$stages[[2]]$param_map$max_iter
```

Comment:
Maybe we leave out `max.iter` just in case people confuse it with `iter`?

Reply:
I was wondering the same thing and I agree with you. 
I didn't find if spark provides the number of iteration, I found just the max.iter and tolerance.
But if I remove max.iter and tol, glance() will have just one statistic, is it ok just print one statistic?

Reply:
I think that's fine.